apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/FirestoreDbBuilderTest.cs

```diff
@@ -45,6 +45,27 @@ public void MaximalConfiguration()
             Assert.IsType<CustomConverter<Guid>>(db.SerializationContext.GetConverter(typeof(Guid)));
         }
 
+        [Fact]
+        public void NamedDatabase()
+        {
+            string databaseId = "testDatabase";
+            string projectId = "testProjectId";
+            FirestoreDb namedDb = new FirestoreDbBuilder
+            {
+                ProjectId = projectId,
+                DatabaseId = databaseId
+            }.Build();
+
+            FirestoreDb db = new FirestoreDbBuilder
+            {
+                ProjectId = projectId,
+                DatabaseId = null
+            }.Build();
+
+            Assert.Equal(databaseId,namedDb.DatabaseId);

```

Comment:
I'd split this one test into two: DefaultDatabase (and don't set the DatabaseId explicitly), and NonDefaultDatabase.